### ISSUE TITLE
Failed to query kubeadm's config map error=configmaps "kubeadm-config" is forbidden

### ISSUE THREAD

**Issue by chetnuy** [projectcalico/calico#4139]
I am using CLUSTER_TYPE = 'k8s, bgp'. I don't need kubeadm when I comment on a section:
```
#Pod CIDR auto-detection on kubeadm needs access to config maps.
#   - apiGroups: [""]
#     resources:
#      - configmaps
#    verbs:
#       - get
```
I cannot start calico and I get an error in the logs of the calico node:
`failed to query kubeadm's config map error = configmaps "kubeadm-config" is forbidden
`
## Your Environment

* Calico version 3.16.4
* Orchestrator version (e.g. kubernetes, mesos, rkt): kubernetes 1.18
* Operating System and version: centos 7
* Link to your project (optional):

**Comment by mgleung** [comment]
Hi @chetnuy, it seems that by commenting out the the permissions for checking configmaps, you are hitting this issue. The message about kubeadm specifically is because Calico is attempting to find a kubeadm config in case it needs it for CIDR auto detection. Is there a specific reason that you are removing that permission?

**Comment by chetnuy** [comment]
Hi, I do not want to give unnecessary rights

**Comment by caseydavenport** [comment]
I would have expected this to be fixed based on this PR that went into v3.16.0: https://github.com/projectcalico/node/pull/541

I wonder if we're getting a different error for some reason.
